### PR TITLE
BOAC-2874, fixes to /admin users view

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2874

### CODE REVIEW

File: boac/api/user_controller.py [hunk 1]
```diff
@@ -56,13 +56,14 @@ def user_profile(uid):
...
-    return tolerant_jsonify(calnet.get_calnet_user_for_csid(app, csid))
+    user = calnet.get_calnet_user_for_csid(app, csid)
+    return tolerant_jsonify(user and _user_by_uid(user['uid']))
 
 
 @app.route('/api/user/by_uid/<uid>')
 @advisor_required
 def user_by_uid(uid):
-    return tolerant_jsonify(calnet.get_calnet_user_for_uid(app, uid))
+    return tolerant_jsonify(_user_by_uid(uid))
```
Comment (johncrossman) [inline boac/api/user_controller.py]: Making `authorized_users_api_feed()` the standard.

File: boac/api/util.py [hunk 2]
```diff
@@ -129,16 +129,17 @@ def authorized_users_api_feed(users, sort_by=None, sort_descending=False):
...
-                },
+            departments.append({
+                'code': m.university_dept.dept_code,
+                'name': m.university_dept.dept_name,
+                'isAdvisor': m.is_advisor,
+                'isDirector': m.is_director,
+                'isScheduler': m.is_scheduler,
+                'automateMembership': m.automate_membership,
             })
+        profile['departments'] = departments
```
Comment (johncrossman) [inline boac/api/util.py]: Makes more sense to serve an array of `dept` objects.

File: src/components/admin/EditUserProfileModal.vue [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+              <b-col><b-form-checkbox id="can-access-canvas-data" v-model="userProfile.canAccessCanvasData"></b-form-checkbox></b-col>
+            </b-row>
+          </b-container>
+        </div>
+        <hr class="mb-1 ml-0 mr-0 mt-1" />
+        <div class="ml-3 mr-2 pt-2">
+          <h3 class="color-grey font-size-18 mb-1">Departments</h3>
+          <div
+            v-if="userProfile.departments.length"
+            v-for="dept in userProfile.departments"
```
Comment (lyttam) [inline src/components/admin/EditUserProfileModal.vue]: @johncrossman Not necessary to use `v-if` with `v-for` and, in fact, it's not recommended:  https://vuejs.org/v2/style-guide/#Avoid-v-if-with-v-for-essential

Comment (johncrossman) [reply src/components/admin/EditUserProfileModal.vue]: Thank you @lyttam  Good tip. Amended.

**Pull request merged**